src/modules/previewpane/powerpreview/settings.cpp

```diff
@@ -89,57 +83,48 @@ namespace PowerPreviewSettings
             {
                 this->UpdateToggleSettingState(newState);
 
-                // If global setting is enable. Add or remove the preview handler otherwise just change the UI and save the updated config.
+                // If global setting is enable. Add or remove the file explorer module otherwise just change the UI and save the updated config.
                 if (enabled)
```

Comment:
@arjunbalgovind:  Do we want to keep this global check in since the user may have it set to false, with no way to undo it.   I _think_ this is causing me problems.  I'm not sure if this was the root cause but I had problems with the following repro.
1.  File Explorer was set to false in the global settings.  (yet i'm pretty sure i had the preview pane working)
2.  I uninstalled the .23 version of pt. Validated that the preview pane wasn't rendering svg or markdown
3.  I built the msi from this branch and installed it. 
4.  I started pt regularly.  *I recieved the warning that some settings couldn't be set* 
5.  I restarted as admin. 
6.  I observed the toggles were enabled but changing them didn't do anything to the settings. 
7.  I manually updated the settings.json so that FileExplorer was true.   This allowed me to get into this if statement, but for some reason explorer was still rendering the svg and md files.  Even with the reg key deleted, and restarting explorer. 

I then uninstalled the msi and reinstalled it with the FileExplorer:true.  and the svg and md files seem to be able to be disabled / enabled correctly.  However the svg thumbnails seem to keep rendering.  (THis also happends even after uninstall for me).

Reply:
@ryanbodrug-microsoft 
Made some tweaks:
1. In the UpdateState method I've changed the check for enabled to `enabled || UseNewSettings()`, where `UseNewSettings` is the method we use for checking the windows version. This ensures that when a user modifies a toggle the setting will get updated for a New Settings user irrespective of the File Explorer state in general settings. This will fix the behavior you observed at step 6 when File Explorer was false.
2. Regarding this
    > I started pt regularly. I recieved the warning that some settings couldn't be set
    I found there was a casing issue in the C++ side for the content of the registry key (which is just the display name of the handler). That is why on first startup it was trying to replace it. Fixed it to match the casing in the MSI project.


I did some testing on a fresh VM and these were my findings:
- For SVG and MD preview, disabling the toggle will instantly disable it. If you have explorer open and an md file is open it will be previewed till you click another file. My guess is that as soon as you click away from the file prevhost unloads the dll, so after that it works as per the settings. So preview pane seems to work perfectly, no reboot/restarting explorer required
- For thumbnail providers, if you enable/disable the setting may/may not get updated immediately if you don't have an explorer window running. This also applies for uninstalling. If you restart explorer in some cases it will accordingly enable or disable the thumbnail. I did some digging and my hunch was right, the reason why it may not go even after restarting explorer is because the thumbnail might still be in the thumbnail cache. I'm not exactly sure under what circumstances does Windows normally updates the thumbnail cache, but I was able to get it to match the setting by clearing the thumbnail cache using the steps in this link https://neosmart.net/wiki/clear-thumbnails-cache/#Method_2-4, i.e. Disk Cleanup ->Thumbnails. I was able to consistently get the thumbnails enabled/disabled by restarting explorer and doing the thumbnail cache cleanup every time. 
Because of the observations I've mentioned above, I was thinking of updating the warning for thumbnail providers to also mention that it may require cleaning up the thumbnail cache. Since most users won't know how to do this, we could add steps in the File Explorer wiki, and add a hyperlink pointing to that.

The issue with this is that I don't think there are any official Microsoft docs which call out that this is the way to cleanup the thumbnail cache (though Disk Cleanup seems like the most legit way to do it). Adding this to the readme might result in us making it "official". What should we do here? (tagging @crutkas for visibility)

Reply:
I've updated the "enable" logic to get called in the constructor for New settings so that we can ignore the general settings value. Its still in enable() for old settings.
As discussed offline I'll make a separate issue for calling out the thumbnail cache issue, and I'll leave the current warning message as it is.